src/Microsoft.IdentityModel.Tokens/SecurityTokenDescriptor.cs

```diff
@@ -12,10 +16,61 @@ namespace Microsoft.IdentityModel.Tokens
     /// </summary>
     public class SecurityTokenDescriptor
     {
+        // TODO: At next major version, remove _audienceSet, _audiencesSet, UseAudiences, Audience, and relevant logic.
+        private bool _audienceSet = false;
+        private bool _audiencesSet = false;
+        private IList<string> _audiences;
+        internal bool UseAudiences => _audiencesSet;
+        internal bool HasAudience => _audiences != null && _audiences.Count > 0;
+        internal string AudiencesJson
```

Comment:
not a big fan of adding more internals

Reply:
Do you have any suggestions on how to achieve the current goal of running both Audience and Audiences flows simultaneously without internals?

The intent is that they are temporary and will be removed when we remove Audience and switch to only Audiences. 

Out of curiosity, what do you not like about adding internals?

Reply:
sorry, I should have been more clear, internals used as friends